### PR TITLE
openssh: update to 9.7p1

### PR DESCRIPTION
**Pull request by artkiver**
###### Type(s)

- [x] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on

macOS 14.4 23E214 arm64
Xcode 15.3 15E204a

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL in commit message? 
- [x] checked your Portfile with `port lint --nitpick`?
- [x] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?
- [x] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

**Comment by herbygillot** [comment]
CI is also running tests.  It looks like it built correctly, but is failing the `port test` phase:

```
 --->  Testing openssh
  DEBUG: Executing org.macports.test (openssh)
  Error: Failed to test openssh: fopen(/opt/local/var/macports/build/_Users_runner_work_macports-ports_macports-ports_ports_net_openssh/openssh/work/destroot/opt/local/libexec/ssh-keysign): permission denied
  DEBUG: Error code: POSIX EACCES {permission denied}
  DEBUG: Backtrace: fopen(/opt/local/var/macports/build/_Users_runner_work_macports-ports_macports-ports_ports_net_openssh/openssh/work/destroot/opt/local/libexec/ssh-keysign): permission denied
      while executing
  "fileIsBinary $fullpath"
      invoked from within
  "fs-traverse -depth fullpath [list [option destpath]] {
          if {[file type $fullpath] ne "file"} {
              continue
          }
          if {[file..."
      (procedure "porttest::test_archs" line 11)
      invoked from within
  "porttest::test_archs"
      (procedure "porttest::test_main" line 5)
      invoked from within
  "$procedure $targetname"
```

**Comment by artkiver** [comment]
> CI is also running tests. It looks like it built correctly, but is failing the `port test` phase:
> 
> ```
>  --->  Testing openssh
>   DEBUG: Executing org.macports.test (openssh)
>   Error: Failed to test openssh: fopen(/opt/local/var/macports/build/_Users_runner_work_macports-ports_macports-ports_ports_net_openssh/openssh/work/destroot/opt/local/libexec/ssh-keysign): permission denied
>   DEBUG: Error code: POSIX EACCES {permission denied}
>   DEBUG: Backtrace: fopen(/opt/local/var/macports/build/_Users_runner_work_macports-ports_macports-ports_ports_net_openssh/openssh/work/destroot/opt/local/libexec/ssh-keysign): permission denied
>       while executing
>   "fileIsBinary $fullpath"
>       invoked from within
>   "fs-traverse -depth fullpath [list [option destpath]] {
>           if {[file type $fullpath] ne "file"} {
>               continue
>           }
>           if {[file..."
>       (procedure "porttest::test_archs" line 11)
>       invoked from within
>   "porttest::test_archs"
>       (procedure "porttest::test_main" line 5)
>       invoked from within
>   "$procedure $targetname"
> ```

Yeah, as I mentioned here: https://trac.macports.org/ticket/69478#ticket I had ignored port test failing locally because I am usually accustomed to that failing when it attempts to netcat to sshd (and I have a firewall running so that stalls out/fails) and in my haste I didn't look at the main.log output in detail before submitting the PR.

The permissions error looks similar to something addressed upstream by OpenSMTPD; and I made mentioned of that in the Trac issue as well as tagged OpenSSH developer @djm-google on the FediVerse, but I have not yet looked into this deeply, so my cursory evaluation may be a bit of a red herring.

**Comment by djm-google** [comment]
ITYM @djmdjm (same person, different role)

File: net/openssh/Portfile [hunk 1]
```diff
@@ -5,7 +5,7 @@ PortSystem          1.0
 PortGroup           compiler_blacklist_versions 1.0
 
 name                openssh
-version             9.6p1
+version             9.7p1
 revision            1
```
Comment (reneeotten) [inline net/openssh/Portfile:9]: `revision` should be reset to `0` when updating the `version`

**Review by reneeotten — CHANGES_REQUESTED** [review]

**Review by artkiver — COMMENTED** [review]
Another glaring oversight by me. Thanks for the catch @reneeotten !

**Comment by artkiver** [comment]
As a "sanity check" of sorts, I checked locally to see if 9.6p1 maybe failed similarly? It seems as if that is the case.

Reverting to the previous Portfile and running port -d -v test, it fails as follows:

`
DEBUG: test phase started at Tue Mar 12 15:43:37 UTC 2024
--->  Testing openssh
DEBUG: Executing org.macports.test (openssh)
Error: Failed to test openssh: fopen(/opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_net_openssh/openssh/work/destroot/opt/local/libexec/ssh-keysign): permission denied
DEBUG: Error code: POSIX EACCES {permission denied}
DEBUG: Backtrace: fopen(/opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_net_openssh/openssh/work/destroot/opt/local/libexec/ssh-keysign): permission denied
    while executing
"fileIsBinary $fullpath"
    invoked from within
"fs-traverse -depth fullpath [list [option destpath]] {
        if {[file type $fullpath] ne "file"} {
            continue
        }
        if {[file..."
    (procedure "porttest::test_archs" line 11)
    invoked from within
"porttest::test_archs"
    (procedure "porttest::test_main" line 5)
    invoked from within
"$procedure $targetname"
Error: See /opt/local/var/macports/logs/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_net_openssh/openssh/main.log for details.
Error: Follow https://guide.macports.org/#project.tickets if you believe there
is a bug.
Error: Processing of port openssh failed
`

I vaguely recall some discussion on the mailing list that MacPorts' tests had changed recently, and I know that whatever CI checks were run when I submitted the PR for 9.6p1 passed OK. (https://github.com/macports/macports-ports/pull/21834) but that was in December of last year, I think before 2.9.1?

Regardless, I am currently wondering if this might be a MacPorts issue rather than one for handling upstream?

**Comment by artkiver** [comment]
I updated the Trac issue as well with output from %make tests run manually from the OpenSSH 9.7p1 tarball, here:
https://trac.macports.org/ticket/69478#comment:2

As you can see from that, it ends with:

`all tests passed`

So, I think this is an issue related to this commit from MacPorts 2.9.0: https://github.com/macports/macports-base/commit/ddc26554e6ed49ce8501069b967284739fa6d34e ?

Namely, "Added a simple default test phase for all ports, which currently
just checks the architectures of installed files. test.run yes
enables the tests defined by the Portfile, as before. It should
always be safe to run port test now, and if it fails it means
a test didn't pass. (jmr in https://github.com/macports/macports-base/commit/ddc26554e6ed49ce8501069b967284739fa6d34e)"

I responded to a similar thread on the mailing list that another developer had encountered with some of the R related ports I guess; since I think this probably is not an upstream issue, but has to do with the "simple default test phase" that MacPorts introduced since I last submitted a PR for OpenSSH in December 2023.

Not too sure how to move forward with this yet, but maybe @jmroot has some useful insights?

Thanks!

**Pull request merged**